File: frontend/user/lib/domain/usecase/order/confirm_delivery_usecase.dart
import 'package:b2205946_duonghuuluan_luanvan/core/error/failure.dart';
import 'package:dartz/dartz.dart';
import 'package:b2205946_duonghuuluan_luanvan/domain/entity/order/order_models.dart';
import 'package:b2205946_duonghuuluan_luanvan/domain/repository/order/order_repository.dart';

class ConfirmDeliveryUseCase {
  final OrderRepository _repo;
  ConfirmDeliveryUseCase(this._repo);
  Future<Either<Failure, OrderOut>> call(int orderId) => _repo.confirmDelivery(orderId);
}

File: backend/app/application/use_case/order/confirm_delivery_usecase.py
from app.domain.repositories.order_repository import OrderRepository


class ConfirmDeliveryUseCase:
    def __init__(self, order_repo: OrderRepository):
        self.order_repo = order_repo

    def execute(self, order_id: int, user_id: int):
        return self.order_repo.confirm_delivery(order_id, user_id)
